mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -171,55 +166,104 @@ public void commitJob(JobContext originalContext) throws IOException {
+   * @param location The location of the table used for loading from the catalog
+   */
+  private void commitTable(JobContext jobContext, String name, String location) {
+    JobConf conf = jobContext.getJobConf();
+    Properties catalogProperties = new Properties();
+    catalogProperties.put(Catalogs.NAME, name);
+    catalogProperties.put(Catalogs.LOCATION, location);
+    Table table = Catalogs.loadTable(conf, catalogProperties);
+
+    long startTime = System.currentTimeMillis();
+    LOG.info("Committing job has started for table: {}, using location: {}",
+        table, generateJobLocation(location, conf, jobContext.getJobID()));
+
+    FileIO io = HiveIcebergStorageHandler.io(conf);
+    List<DataFile> dataFiles = dataFiles(location, jobContext, io, true);
+
+    if (dataFiles.size() > 0) {
+      // Appending data files to the table
+      AppendFiles append = table.newAppend();
+      dataFiles.forEach(append::appendFile);
+      append.commit();
+      LOG.info("Commit took {} ms for table: {} with {} file(s)", System.currentTimeMillis() - startTime, table,
+          dataFiles.size());
+      LOG.debug("Added files {}", dataFiles);
+    } else {
+      LOG.info("Commit took {} ms for table: {} with no new files", System.currentTimeMillis() - startTime, table);
+    }
+  }
+
   /**
    * Cleans up the jobs temporary location.
    * @param jobContext The job context
    * @throws IOException if there is a failure deleting the files
    */
   private void cleanup(JobContext jobContext) throws IOException {
-    String location = generateJobLocation(jobContext.getJobConf(), jobContext.getJobID());
-    LOG.info("Cleaning for job: {} on location: {}", jobContext.getJobID(), location);
-
-    // Remove the job's temp directory recursively.
-    // Intentionally used foreach on a single item. Using the Tasks API here only for the retry capability.
-    Tasks.foreach(location)
-        .retry(3)
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.debug("Failed on to remove directory {} on cleanup job", file, exc))
-        .run(file -> {
-          Path toDelete = new Path(file);
-          FileSystem fs = Util.getFs(toDelete, jobContext.getJobConf());
-          fs.delete(toDelete, true);
-        }, IOException.class);
+    JobConf jobConf = jobContext.getJobConf();
+
+    LOG.info("Cleaning for job {} started", jobContext.getJobID());
+    Map<String, String> outputs = SerializationUtil.deserializeFromBase64(jobConf.get(InputFormatConfig.OUTPUT_TABLES));
+
+    for (String location : outputs.keySet()) {
+      String tableLocation = generateJobLocation(location, jobConf, jobContext.getJobID());
+      LOG.info("Cleaning location: {}", tableLocation);
+
+      // Remove the job's temp directory recursively.
+      // Intentionally used foreach on a single item. Using the Tasks API here only for the retry capability.
+      Tasks.foreach(tableLocation)
```

Comment:
Now that we're cleaning up multiple locations, can we take advantage of the parallelism here?

Reply:
Good idea, applied to most of the places